### ISSUE TITLE
Too Many Open Files Error on Galasactl runs download

### ISSUE THREAD

**Issue by Tom-Slattery** [galasa-dev/projectmanagement#1732]
## Describe the bug
Galasactl (v 0.30.0) gets "too many open files" error on downloading artifacts from a run onto a devcontainer. Possibly not closing files?

```
root ➜ /workspaces (devcontainer) $ galasactl runs download --name U38
GAL2501I: Downloaded 54 artifacts to folder 'U38'
GAL1042E: Failed to open file 'U38/artifacts/sem/discard/GALJGZS4/JOB97121_JES2_JESYSMSG' for writing. Reason is 'open U38/artifacts/sem/discard/GALJGZS4/JOB97121_JES2_JESYSMSG: too many open files'. Check that you have permissions to write to that folder and file, and that there is enough disk space available and try again.
```

## Steps to reproduce
1. Run galasa test in 0.31.0 ecosystem using 0.30.0 galasactl 
2. galasactl runs download 

## Expected behavior
Should download all artifacts

**Comment by techcobweb** [comment]
runsDownload is allocating a writer attached to a file, but not closing it.
The interface should return a WriteCloser instead, so the file can be closed once the artifact stuff is written.
If it's closed on a defer, then we can guarantee that it will be closed-out also.

**Comment by eamansour** [comment]
Fix delivered and will be released in 0.32.0
